From 4822ceb9b15427d4ef138b3655d10c114fc795f9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:23 -0400
Subject: [tools] perf/symbols: Fix random fd closing with no libelf

Message-id: <1408953210-26343-76-git-send-email-jolsa@redhat.com>
Patchwork-id: 88804
O-Subject: [PATCH RHEL7.1 BZ1133083 075/322] perf symbols: Fix random fd closing with no libelf
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 779e24e2c777bffa9a6e3d5e821859e67008c98b
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Wed Dec 4 16:23:01 2013 +0200

description
===========
When built without libelf, perf tools was failing to initialize a file
descriptor, but nevertheless closing it.  That sometimes resulted in the
output being truncated because the stdout file descriptor got closed.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/symbol-minimal.c b/tools/perf/util/symbol-minimal.c
index 2d2dd05..ac7070a 100644
--- a/tools/perf/util/symbol-minimal.c
+++ b/tools/perf/util/symbol-minimal.c
@@ -253,6 +253,7 @@ int symsrc__init(struct symsrc *ss, struct dso *dso __maybe_unused,
  if (!ss->name)
   goto out_close;
 
+ ss->fd = fd;
  ss->type = type;
 
  return 0;
-- 
1.7.1